src/db.rs

```diff
@@ -142,28 +147,40 @@ pub fn delete_connection(uuid: Uuid) -> Result<Uuid> {
     Ok(uuid)
 }
 
-pub fn get_connection(uuid: Uuid) -> Result<Connection> {
+pub fn get_connection(uuid: Uuid) -> Result<(Connection, String)> {
     let mut dbconn = get_dbconn()?;
 
     let uuidstr = uuid.as_hyphenated().to_string();
-    let mqstr = "SELECT policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest FROM conn_bundle WHERE id = ?";
+    let mqstr = "SELECT policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest, symkey FROM conn_bundle WHERE id = ?";
 
     let conres = dbconn.exec_map(
         mqstr,
         (uuidstr,),
-        |(policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest)| {
-            Connection {
-                policy,
-                fw_api_major,
-                fw_api_minor,
-                fw_build_id,
-                launch_description,
-                fw_digest,
-            }
+        |(
+            policy,
+            fw_api_major,
+            fw_api_minor,
+            fw_build_id,
+            launch_description,
+            fw_digest,
+            key_b64,
+        )|
+         -> (Connection, String) {
+            (
+                Connection {
+                    policy,
+                    fw_api_major,
+                    fw_api_minor,
+                    fw_build_id,
+                    launch_description,
+                    fw_digest,
+                },
+                key_b64,
+            )
```

Comment:
You return a pair: `(Connection, String)`. Have you considered adding `key_b64: Option<String>` to the Connection struct?

Reply:
The new report secret type gives back a signed version of the connection. I don't think we want the symmetric key in that report, so I left it out of the connection struct, which currently only contains values provided by the guest.

Reply:
OK, make sense; keep them separated.